filters/private/hexer/H3grid.hpp

```diff
@@ -0,0 +1,106 @@
+#pragma once
+
+#if WINDOWS
+#undef min
+#undef max
+#endif // WINDOWS
+
+#include <h3api.h>
+
+#include "BaseGrid.hpp"
+
+namespace hexer
+{
+
+using DirEdge = H3Index;
+
+class PDAL_DLL H3Grid : public BaseGrid
+{
+public:
+    H3Grid(int dense_limit)
+        : BaseGrid{dense_limit}, m_res{-1}, 
+            m_minI{std::numeric_limits<int>::max()}, m_origin{0}
+    {}
+    H3Grid(int res, int dense_limit)
+        : BaseGrid{dense_limit}, m_res{res}, 
+            m_minI{std::numeric_limits<int>::max()}, m_origin{0}
+    {}
+    ~H3Grid();
+
+    H3Index ij2h3(HexId ij)
+        {   H3Index h3;
+            if (PDALH3localIjToCell(m_origin, &ij, 0, &h3) != E_SUCCESS) {
+                std::ostringstream oss;
+                oss << "Can't convert IJ (" << ij.i <<
+                    ", " << ij.j <<") to H3Index.";
+                throw hexer_error(oss.str());
+            }
+            return h3;  }
+
+    // Convert H3 index to IJ coordinates
+    HexId h32ij(H3Index h3)
+        {   HexId ij;
+            if (PDALH3cellToLocalIj(m_origin, h3, 0, &ij) != E_SUCCESS) {
+                std::ostringstream oss;
+                oss << "Can't convert H3 index " << h3 <<
+                    " to IJ.";
+                throw hexer_error(oss.str());
+            }
+            return ij;  }
+
+    Point findPoint(Segment& s);
+
+    void addXY(double& x, double& y)
+        {
+          Point p{PDALH3degsToRads(x), PDALH3degsToRads(y)};
+          addPoint(p);        
+        }
+    double height()
```

Comment:
I'm calculating 2*apothem (what height represents in HexGrid) from the length of one side. It could be wrong, I don't know. A couple metadata elements needed it, originally was using it when I was doing smoothing for H3 boundaries but now it only is used for avg_pt_per_sq_unit.

Reply:
OK. I messed up what p1 and p2 were. But you can certainly eliminate the multiplication and subsequent division by 2.